### PR TITLE
Document monitor threshold overrides via annotations

### PR DESCRIPTION
**Pull request by aacevedoosorio**

### CODE REVIEW

File: use/alerting/k8s-override-monitor-arguments.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+---
+description: StackState Kubernetes Troubleshooting
+---
+
+# Override monitor threshold arguments via kubernetes annotations
+
+## Overview
+
+StackState provides [monitors out of the box](/use/alerting/k8s-monitors.md), which provide monitoring on common issues that can occur in a Kubernetes cluster. Those monitors work with certain default arguments that suit most of the use cases but sometimes we need to adapt its behaviour by overriding some of such default arguments  like `threshold` or `failureState`.
```
[the remark this answers is not in the document]
Comment (craffit) [reply use/alerting/k8s-override-monitor-arguments.md:9]: like vale says:)

File: use/alerting/k8s-add-monitors-cli.md [hunk 2]
```diff
@@ -39,7 +39,7 @@ nodes:
     urnTemplate: "urn:kubernetes:/${kube_cluster_name}:${namespace}:service/${endpoint}"
     titleTemplate: 
   description: 
-  function: {{ get "urn:stackpack:kubernetes-v2:shared:monitor-function:threshold"  }}
+  function: {{ get "urn:stackpack:common:monitor-function:threshold"  }}
```
Comment (craffit) [inline use/alerting/k8s-add-monitors-cli.md:42]: nice find

File: use/alerting/kubernetes-monitors.md [hunk 3]
```diff
@@ -34,12 +44,33 @@ It is important to monitor the usage of Persistent Volume Claims (PVCs) in your
...
...ments, they may consume more PIDs than expected, resulting in node PID pressure. The second reason is the presence of PID-intensive applications or processes. Some workloads or applications have higher demands for process identification, leading to increased PID utilization on the node. If multiple pods or containers with significant PID requirements are scheduled on the same node without proper resource allocation, it can cause PID pressure. To address node PID pressure, it is important to review and adjust resource requests and limits for containers to ensure they align with the actual PID needs of the applications. Monitoring and optimizing PID usage within the applications themselves can also help reduce PID consumption. Additionally, considering horizontal pod autoscaling can dynamically scale the number of pods based on PID utilization. Regular monitoring, analysis of PID-related metrics, and proactive allocation of PID resources are crucial for maintaining a healthy state of PID usage on Kubernetes nodes. It is essential to understand the specific requirements of your workloads and adjust resource allocation accordingly to prevent PID pressure and ensure optimal performance.
```
Comment (craffit) [inline use/alerting/kubernetes-monitors.md:59]: is this copy-paste from the descriptions? if so, no more review needed

File: use/alerting/k8s-override-monitor-arguments.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      }
+```
+
+## What monitors allow overriding arguments?
+* [Available service endpoints](/use/alerting/kubernetes-monitors.md#available-service-endpoints)
+* [Node Disk Pressure](/use/alerting/kubernetes-monitors.md#node-disk-pressure)
+* [Node Memory Pressure](/use/alerting/kubernetes-monitors.md#node-memory-pressure)
+* [Node PID Pressure](/use/alerting/kubernetes-monitors.md#node-pid-pressure)
+* [Node Readiness](/use/alerting/kubernetes-monitors.md#node-readiness)
+* [Out of memory for containers](/use/alerting/kubernetes-monitors.md#out-of-memory-for-containers)
```
Comment (craffit) [inline use/alerting/k8s-override-monitor-arguments.md:59]: Maybe a last short section how to do an override on a custom monitor, to have that example in aswell

File: use/alerting/kubernetes-monitors.md (hunk 3 above)
Comment (aacevedoosorio) [reply use/alerting/kubernetes-monitors.md:59]: Yup. I took it directly from the descriptions on master

**Review by craffit — APPROVED** [review]
Approved, looking good

**Pull request merged**